roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -55,59 +65,85 @@ def required_images(self, task_vars):
-            if cmd.get("images", None):
-                found.add(image)
+        """Filter a list of images and return those available locally."""
+        return [
+            image for image in required_images
+            if self.is_image_local(image, task_vars)
+        ]
 
-        return found
+    def is_image_local(self, image, task_vars):
+        result = self.module_executor("docker_image_facts", {"name": image}, task_vars)
+        if result.get("failed", False):
+            return False
+
+        return bool(result.get("images", []))
 
     def known_docker_registries(self, task_vars):
-        docker_info = self.module_executor("docker_info", {}, task_vars).get("info", "")
+        result = self.module_executor("docker_info", {}, task_vars)
+
+        # do test on module exception - see if this line is useful
+        if result.get("failed", False):
+            return []
+
+        docker_info = result.get("info", "")
         return [registry.get("Name", "") for registry in docker_info.get("Registries", {})]
 
-    def available_images(self, missing_images, registries, task_vars):
-        """ Inspect existing images using Skopeo and return all images successfully inspected """
-        missing_images_copy = list(missing_images)
-        registries_copy = list(registries)
+    def available_images(self, images, registries, task_vars):
+        """Inspect existing images using Skopeo and return all images successfully inspected."""
+        return [
+            image for image in images
+            if self.is_image_available(image, registries, task_vars)
+        ]
+
+    def is_image_available(self, image, registries, task_vars):
+        for registry in registries:
+            if self.is_available_skopeo_image(image, registry, task_vars):
+                return True
 
-        skopeo_update = self.update_skopeo_image(task_vars)
-        changed = skopeo_update.get("changed", False)
+        return False
 
-        # exit early if Skopeo update fails
-        if skopeo_update.get("failed", False):
-            return missing_images, changed
+    def is_available_skopeo_image(self, image, registry, task_vars):
+        """Uses Skopeo to determine if required image exists in a given registry.
 
-        while registries_copy and missing_images_copy:
-            registry = registries_copy.pop(0)
-            # pylint: disable=redefined-variable-type
-            missing_images_copy = self.missing_skopeo_images(registry, missing_images_copy, task_vars)
+         Returns the set of images not found in the given registry."""
 
-        return set(missing_images) - set(missing_images_copy), changed
+        cmd_str = "skopeo inspect docker://{registry}/{image}".format(
+            registry=registry,
+            image=image,
+        )
+
+        # check how container name collisions are handled
+        # make sure container is removed by module
```

Comment:
Yes, I verified (with `docker ps`) that in an enterprise install in an aws ec2 instance, the container is not still there after the check completes